lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -1324,7 +1324,8 @@ struct FIRRTLLowering : public FIRRTLVisitor<FIRRTLLowering, LogicalResult> {
   template <typename ResultOpType, typename... CtorArgTypes>
   LogicalResult setLoweringTo(Operation *orig, CtorArgTypes... args);
   void emitRandomizePrologIfNeeded();
-  void initializeRegister(Value reg);
+  void initializeRegister(Value reg, llvm::Optional<std::pair<Value, Value>>
+                                         asyncRegResetInitPair = llvm::None);
```

Comment:
The PR could avoid having to cache this and putting the check inside the `initializeRegister` member function.  I.e., it can just do the lookup to see if the register is async (and possibly this would make sense as an `isAsync` helper method) and then using the `resetSignal`/`resetValue` already on the op.

Reply:
`reg` is `sv::RegOp` here so we don't know whether `reg` is async or not anymore.
I believe we have to pass lowered `resetSignal/resetValue` from that reason.

Reply:
Sounds good to me.